### PR TITLE
Update controller images

### LINKED ISSUE #336: VolumeSnapshotContent with Retain policy cannot be deleted
I created a VolumeSnapshot dynamically with deletionPolicy as "Retain". Then I deleted the VolumeSnapshot, it got deleted but the corresponding VolumeSnapshotContent remained, which is expected.
Now I explicitly tried to delete the VolumeSnapshotContent but it never gets deleted.

One observation is that the "volumesnapshotcontent-bound-protection" finalizer is still there on the object. The code to remove this finalizer is never getting invoked in case of Retain policy. The culprit seems to be sidecar-controller/snapshot_controller.shouldDelete(). The only conditions when it can return true are-

        if content.Spec.Source.SnapshotHandle != nil && content.Spec.VolumeSnapshotRef.UID == "" 
and 

	if metav1.HasAnnotation(content.ObjectMeta, utils.AnnVolumeSnapshotBeingDeleted) 

Neither of these are true in case of the VolumeSnapshotContent in concern and hence the finalizer removal code is never invoked. However if I manually delete the finalizer by editing the object, the VolumeSnapshotContent gets deleted.

Snapshotter image version used: v2.1.1
Kubernetes version: v1.18.2

### PR DESCRIPTION
**Pull request by Kartik494**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #336 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml [hunk 1]
```diff
@@ -84,7 +84,7 @@ spec:
             - name: socket-dir
               mountPath: /csi
         - name: csi-snapshotter
-          image: quay.io/k8scsi/csi-snapshotter:v2.0.1
+          image: quay.io/k8scsi/csi-snapshotter:v2.1.1
```
Comment (xing-yang) [inline deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml:87]: In the master, can you just update the image to canary?
In the stable branches, can up submit separate PRs to update image to specific versions?

Comment (Kartik494) [reply deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml]: @xing-yang i have addressed the changes.PTAL

**Comment by xing-yang** [comment]
@Kartik494 @ggriffiths we should use images from the new repos now:
k8s.gcr.io/sig-storage/csi-snapshotter:v2.1.1

**Comment by xing-yang** [comment]
Never mind.  We don't have canary images in the new repos.  So for now, let's continue to use canary images in quay.io for master until we have something else.
